core/src/test/resources/sql/join.iq

```diff
@@ -200,7 +200,7 @@ EnumerableCalc(expr#0..1=[{inputs}], DEPTNO=[$t1], ENAME=[$t0])
     EnumerableHashJoin(condition=[=($2, $4)], joinType=[inner])
       EnumerableCalc(expr#0..7=[{inputs}], expr#8=[10], expr#9=[+($t7, $t8)], proj#0..1=[{exprs}], $f8=[$t9])
         EnumerableTableScan(table=[[scott, EMP]])
-      EnumerableCalc(expr#0..3=[{inputs}], expr#4=[+($t1, $t3)], expr#5=[CAST($t4):INTEGER], DEPTNO=[$t1], $f16=[$t5])
+      EnumerableCalc(expr#0..3=[{inputs}], expr#4=[+($t3, $t1)], expr#5=[CAST($t4):INTEGER], DEPTNO=[$t3], $f16=[$t5])
```

Comment:
Do you know why this was impacted if there is no EnumerableMergeJoin involved?

Reply:
Changes in EnumerableMergeJoinRule proposed in this PR will have an impact on the matching order of rules(The transformation logic in EnumerableMergeJoinRule is changed). When computing the `best` of of a `RelSubset`, it's updated only when a plan with lower cost happens.
As we can see, the plan here comes from a plan after join reorder. If it's generated before a Join without reorder, such change could happen.

Reply:
Thanks for the explanation, @jinxing64